# **PURPOSE**
To analyze the dataset to deliver insights extracted

### **FIRST IMPORTING ENV**

In [3]:
!pip install numpy pandas matplotlib seaborn sklearn

In [12]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
from statistics import *


 **LOADING THE DATA**

In [6]:
Path_to_data = "../data/telcom.csv"

In [7]:
#Read data from csv file
db = pd.read_csv(Path_to_data, na_values=['?', None])

#Display data
db.head()

Bearer Id            Start  Start ms              End  End ms  \
0  1.311448e+19   4/4/2019 12:01     770.0  4/25/2019 14:35   662.0   
1  1.311448e+19   4/9/2019 13:04     235.0   4/25/2019 8:15   606.0   
2  1.311448e+19   4/9/2019 17:42       1.0  4/25/2019 11:58   652.0   
3  1.311448e+19   4/10/2019 0:31     486.0   4/25/2019 7:36   171.0   
4  1.311448e+19  4/12/2019 20:10     565.0  4/25/2019 10:40   954.0   

   Dur. (ms)          IMSI  MSISDN/Number          IMEI  \
0  1823652.0  2.082014e+14   3.366496e+10  3.552121e+13   
1  1365104.0  2.082019e+14   3.368185e+10  3.579401e+13   
2  1361762.0  2.082003e+14   3.376063e+10  3.528151e+13   
3  1321509.0  2.082014e+14   3.375034e+10  3.535661e+13   
4  1089009.0  2.082014e+14   3.369980e+10  3.540701e+13   

      Last Location Name  ...  Youtube DL (Bytes)  Youtube UL (Bytes)  \
0  9.16456699548519E+015  ...          15854611.0           2501332.0   
1                L77566A  ...          20247395.0          19111729.0   
2                D42335A  ...          19725661.0          14699576.0   
3                T21824A  ...          21388122.0          15146643.0   
4                D88865A  ...          15259380.0          18962873.0   

   Netflix DL (Bytes)  Netflix UL (Bytes)  Gaming DL (Bytes)  \
0           8198936.0           9656251.0        278082303.0   
1          18338413.0          17227132.0        608750074.0   
2          17587794.0           6163408.0        229584621.0   
3          13994646.0           1097942.0        799538153.0   
4          17124581.0            415218.0        527707248.0   

   Gaming UL (Bytes)  Other DL (Bytes)  Other UL (Bytes)  Total UL (Bytes)  \
0         14344150.0       171744450.0         8814393.0        36749741.0   
1          1170709.0       526904238.0        15055145.0        53800391.0   
2           395630.0       410692588.0         4215763.0        27883638.0   
3         10849722.0       749039933.0        12797283.0        43324218.0   
4          3529801.0       550709500.0        13910322.0        38542814.0   

   Total DL (Bytes)  
0       308879636.0  
1       653384965.0  
2       279807335.0  
3       846028530.0  
4       569138589.0  

[5 rows x 55 columns]

In [26]:
# column names
db.columns.tolist()

['Bearer Id',
 'Start',
 'Start ms',
 'End',
 'End ms',
 'Dur. (ms)',
 'IMSI',
 'MSISDN/Number',
 'IMEI',
 'Last Location Name',
 'Avg RTT DL (ms)',
 'Avg RTT UL (ms)',
 'Avg Bearer TP DL (kbps)',
 'Avg Bearer TP UL (kbps)',
 'TCP DL Retrans. Vol (Bytes)',
 'TCP UL Retrans. Vol (Bytes)',
 'DL TP < 50 Kbps (%)',
 '50 Kbps < DL TP < 250 Kbps (%)',
 '250 Kbps < DL TP < 1 Mbps (%)',
 'DL TP > 1 Mbps (%)',
 'UL TP < 10 Kbps (%)',
 '10 Kbps < UL TP < 50 Kbps (%)',
 '50 Kbps < UL TP < 300 Kbps (%)',
 'UL TP > 300 Kbps (%)',
 'HTTP DL (Bytes)',
 'HTTP UL (Bytes)',
 'Activity Duration DL (ms)',
 'Activity Duration UL (ms)',
 'Dur. (ms).1',
 'Handset Manufacturer',
 'Handset Type',
 'Nb of sec with 125000B < Vol DL',
 'Nb of sec with 1250B < Vol UL < 6250B',
 'Nb of sec with 31250B < Vol DL < 125000B',
 'Nb of sec with 37500B < Vol UL',
 'Nb of sec with 6250B < Vol DL < 31250B',
 'Nb of sec with 6250B < Vol UL < 37500B',
 'Nb of sec with Vol DL < 6250B',
 'Nb of sec with Vol UL < 1250B',
 'Socia

EXPLORING THE DATAFRAME

In [ ]:
# checking data types
db.info()

In [8]:
# number of data points
print(f" There are {db.shape[0]} rows and {db.shape[1]} columns")

 There are 150001 rows and 55 columns


In [ ]:
# Finding mode for manufacturer
db.mode()['Handset Manufacturer'][0]

In [ ]:
# Finding mode for handset type
db.mode()['Handset Type'][0]

In [13]:
# selecting columns of interest
manufacturer = db['Handset Manufacturer']
manufacturer.head()
manufacturer.describe()

count     149429
unique       170
top        Apple
freq       59565
Name: Handset Manufacturer, dtype: object

In [ ]:
# Getting top 10 handsets used by customers
n = 10
db['Handset Type'].value_counts()[:n].index.tolist()

In [17]:
# Getting the top 3 handset manufacturers
n = 3
manufacturer.value_counts()[:n].index.tolist()

['Apple', 'Samsung', 'Huawei']

In [14]:
mobitype = db['Handset Type']
mobitype.head()
mobitype.describe()

count               149429
unique                1396
top       Huawei B528S-23A
freq                 19752
Name: Handset Type, dtype: object

In [15]:
# Using logic on rows to select only rows with apple as manufacturer
apple = db[manufacturer == 'Apple']
apple.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 59565 entries, 22 to 149998
Data columns (total 55 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   Bearer Id                                 59440 non-null  float64
 1   Start                                     59565 non-null  object 
 2   Start ms                                  59565 non-null  float64
 3   End                                       59565 non-null  object 
 4   End ms                                    59565 non-null  float64
 5   Dur. (ms)                                 59565 non-null  float64
 6   IMSI                                      59565 non-null  float64
 7   MSISDN/Number                             59464 non-null  float64
 8   IMEI                                      59565 non-null  float64
 9   Last Location Name                        59405 non-null  object 
 10  Avg RTT DL (ms)                 

In [16]:
# Getting the top 5 handset types in apple
n = 5
apple['Handset Type'].value_counts()[:n].index.tolist()

['Apple iPhone 6S (A1688)',
 'Apple iPhone 6 (A1586)',
 'Apple iPhone 7 (A1778)',
 'Apple iPhone Se (A1723)',
 'Apple iPhone 8 (A1905)']

In [18]:
# Using logic on rows to select only rows with samsung as manufacturer
samsung = db[manufacturer == 'Samsung']
samsung.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 40839 entries, 0 to 149994
Data columns (total 55 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   Bearer Id                                 40689 non-null  float64
 1   Start                                     40839 non-null  object 
 2   Start ms                                  40839 non-null  float64
 3   End                                       40839 non-null  object 
 4   End ms                                    40839 non-null  float64
 5   Dur. (ms)                                 40839 non-null  float64
 6   IMSI                                      40839 non-null  float64
 7   MSISDN/Number                             40579 non-null  float64
 8   IMEI                                      40839 non-null  float64
 9   Last Location Name                        40652 non-null  object 
 10  Avg RTT DL (ms)                  

In [57]:
# Getting the top 5 handset types in samsung
n = 5
samsung['Handset Type'].value_counts()[:n].index.tolist()

['Samsung Galaxy S8 (Sm-G950F)',
 'Samsung Galaxy A5 Sm-A520F',
 'Samsung Galaxy J5 (Sm-J530)',
 'Samsung Galaxy J3 (Sm-J330)',
 'Samsung Galaxy S7 (Sm-G930X)']

In [20]:
# Using logic on rows to select only rows with huawei as manufacturer
huawei = db[manufacturer == 'Huawei']
huawei.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 34423 entries, 6 to 149999
Data columns (total 55 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   Bearer Id                                 34353 non-null  float64
 1   Start                                     34423 non-null  object 
 2   Start ms                                  34423 non-null  float64
 3   End                                       34423 non-null  object 
 4   End ms                                    34423 non-null  float64
 5   Dur. (ms)                                 34423 non-null  float64
 6   IMSI                                      34423 non-null  float64
 7   MSISDN/Number                             34366 non-null  float64
 8   IMEI                                      34423 non-null  float64
 9   Last Location Name                        34326 non-null  object 
 10  Avg RTT DL (ms)                  

In [21]:
# Getting the top 5 handset types in huawei
n = 5
huawei['Handset Type'].value_counts()[:n].index.tolist()

['Huawei B528S-23A',
 'Huawei E5180',
 'Huawei P20 Lite Huawei Nova 3E',
 'Huawei P20',
 'Huawei Y6 2018']

INTEPRETATION AND RECOMMENDATION

The top 3 phone manufactures by users are Apple(59565), Samsung(40839), and Huawei(34423). Marketing campaigns can therefore target these users and especially with the following handset types: 
Apple iPhone 6S (A1688), Apple iPhone 6 (A1586), Apple iPhone 7 (A1778), Apple iPhone Se (A1723), Apple iPhone 8 (A1905), Samsung Galaxy S8 (Sm-G950F), Samsung Galaxy A5 Sm-A520F, Samsung Galaxy J5 (Sm-J530), Samsung Galaxy J3 (Sm-J330), Samsung Galaxy S7 (Sm-G930X), Huawei B528S-23A, Huawei E5180, Huawei P20 Lite Huawei Nova 3E, Huawei P20, Huawei Y6 2018.

ASSIGNMENT 1.1

Transforming the Data (Cleaning, Missing Values)

In [23]:
# summary statistics
db.describe()

Bearer Id       Start ms         End ms     Dur. (ms)          IMSI  \
count  1.490100e+05  150000.000000  150000.000000  1.500000e+05  1.494310e+05   
mean   1.013887e+19     499.188200     498.800880  1.046086e+05  2.082016e+14   
std    2.893173e+18     288.611834     288.097653  8.103762e+04  2.148809e+10   
min    6.917538e+18       0.000000       0.000000  7.142000e+03  2.040471e+14   
25%    7.349883e+18     250.000000     251.000000  5.744050e+04  2.082014e+14   
50%    7.349883e+18     499.000000     500.000000  8.639900e+04  2.082015e+14   
75%    1.304243e+19     749.000000     750.000000  1.324302e+05  2.082018e+14   
max    1.318654e+19     999.000000     999.000000  1.859336e+06  2.140743e+14   

       MSISDN/Number          IMEI  Avg RTT DL (ms)  Avg RTT UL (ms)  \
count   1.489350e+05  1.494290e+05    122172.000000    122189.000000   
mean    4.188282e+10  4.847455e+13       109.795706        17.662883   
std     2.447443e+12  2.241637e+13       619.782739        84.793524   
min     3.360100e+10  4.400152e+11         0.000000         0.000000   
25%     3.365130e+10  3.546071e+13        32.000000         2.000000   
50%     3.366371e+10  3.572201e+13        45.000000         5.000000   
75%     3.368349e+10  8.611970e+13        70.000000        15.000000   
max     8.823971e+14  9.900120e+13     96923.000000      7120.000000   

       Avg Bearer TP DL (kbps)  ...  Youtube DL (Bytes)  Youtube UL (Bytes)  \
count            150000.000000  ...        1.500010e+05        1.500010e+05   
mean              13300.045927  ...        1.163407e+07        1.100941e+07   
std               23971.878541  ...        6.710569e+06        6.345423e+06   
min                   0.000000  ...        5.300000e+01        1.050000e+02   
25%                  43.000000  ...        5.833501e+06        5.517965e+06   
50%                  63.000000  ...        1.161602e+07        1.101345e+07   
75%               19710.750000  ...        1.744852e+07        1.651556e+07   
max              378160.000000  ...        2.325910e+07        2.201196e+07   

       Netflix DL (Bytes)  Netflix UL (Bytes)  Gaming DL (Bytes)  \
count        1.500010e+05        1.500010e+05       1.500010e+05   
mean         1.162685e+07        1.100175e+07       4.220447e+08   
std          6.725218e+06        6.359490e+06       2.439675e+08   
min          4.200000e+01        3.500000e+01       2.516000e+03   
25%          5.777156e+06        5.475981e+06       2.104733e+08   
50%          1.164222e+07        1.099638e+07       4.234081e+08   
75%          1.747048e+07        1.650727e+07       6.331742e+08   
max          2.325919e+07        2.201196e+07       8.434419e+08   

       Gaming UL (Bytes)  Other DL (Bytes)  Other UL (Bytes)  \
count       1.500010e+05      1.500010e+05      1.500010e+05   
mean        8.288398e+06      4.211005e+08      8.264799e+06   
std         4.782700e+06      2.432050e+08      4.769004e+06   
min         5.900000e+01      3.290000e+03      1.480000e+02   
25%         4.128476e+06      2.101869e+08      4.145943e+06   
50%         8.291208e+06      4.218030e+08      8.267071e+06   
75%         1.243162e+07      6.316918e+08      1.238415e+07   
max         1.655879e+07      8.434425e+08      1.655882e+07   

       Total UL (Bytes)  Total DL (Bytes)  
count      1.500000e+05      1.500000e+05  
mean       4.112121e+07      4.546434e+08  
std        1.127639e+07      2.441429e+08  
min        2.866892e+06      7.114041e+06  
25%        3.322201e+07      2.431068e+08  
50%        4.114331e+07      4.558411e+08  
75%        4.903424e+07      6.657055e+08  
max        7.833131e+07      9.029696e+08  

[8 rows x 50 columns]

In [22]:
# checking data types
db.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150001 entries, 0 to 150000
Data columns (total 55 columns):
 #   Column                                    Non-Null Count   Dtype  
---  ------                                    --------------   -----  
 0   Bearer Id                                 149010 non-null  float64
 1   Start                                     150000 non-null  object 
 2   Start ms                                  150000 non-null  float64
 3   End                                       150000 non-null  object 
 4   End ms                                    150000 non-null  float64
 5   Dur. (ms)                                 150000 non-null  float64
 6   IMSI                                      149431 non-null  float64
 7   MSISDN/Number                             148935 non-null  float64
 8   IMEI                                      149429 non-null  float64
 9   Last Location Name                        148848 non-null  object 
 10  Avg RTT DL (ms)     

In [29]:
df_clean.columns.tolist()

['Bearer Id',
 'Start',
 'Start ms',
 'End',
 'End ms',
 'Dur. (ms)',
 'IMSI',
 'MSISDN/Number',
 'IMEI',
 'Last Location Name',
 'Avg RTT DL (ms)',
 'Avg RTT UL (ms)',
 'Avg Bearer TP DL (kbps)',
 'Avg Bearer TP UL (kbps)',
 'DL TP < 50 Kbps (%)',
 '50 Kbps < DL TP < 250 Kbps (%)',
 '250 Kbps < DL TP < 1 Mbps (%)',
 'DL TP > 1 Mbps (%)',
 'UL TP < 10 Kbps (%)',
 '10 Kbps < UL TP < 50 Kbps (%)',
 '50 Kbps < UL TP < 300 Kbps (%)',
 'UL TP > 300 Kbps (%)',
 'Activity Duration DL (ms)',
 'Activity Duration UL (ms)',
 'Dur. (ms).1',
 'Handset Manufacturer',
 'Handset Type',
 'Nb of sec with Vol DL < 6250B',
 'Nb of sec with Vol UL < 1250B',
 'Social Media DL (Bytes)',
 'Social Media UL (Bytes)',
 'Google DL (Bytes)',
 'Google UL (Bytes)',
 'Email DL (Bytes)',
 'Email UL (Bytes)',
 'Youtube DL (Bytes)',
 'Youtube UL (Bytes)',
 'Netflix DL (Bytes)',
 'Netflix UL (Bytes)',
 'Gaming DL (Bytes)',
 'Gaming UL (Bytes)',
 'Other DL (Bytes)',
 'Other UL (Bytes)',
 'Total UL (Bytes)',
 'Total DL (By

In [24]:
# how many missing values exist or better still what is the % of missing values in the dataset?
def percent_missing(df):

    # Calculate total number of cells in dataframe
    totalCells = np.product(df.shape)

    # Count number of missing values per column
    missingCount = df.isnull().sum()

    # Calculate total number of missing values
    totalMissing = missingCount.sum()

    # Calculate percentage of missing values
    print("The Telcom dataset contains", round(((totalMissing/totalCells) * 100), 2), "%", "missing values.")

percent_missing(db)

The Telcom dataset contains 12.5 % missing values.


In [10]:
# Checking for missing values
db.isna().sum()

Bearer Id                                      991
Start                                            1
Start ms                                         1
End                                              1
End ms                                           1
Dur. (ms)                                        1
IMSI                                           570
MSISDN/Number                                 1066
IMEI                                           572
Last Location Name                            1153
Avg RTT DL (ms)                              27829
Avg RTT UL (ms)                              27812
Avg Bearer TP DL (kbps)                          1
Avg Bearer TP UL (kbps)                          1
TCP DL Retrans. Vol (Bytes)                  88146
TCP UL Retrans. Vol (Bytes)                  96649
DL TP < 50 Kbps (%)                            754
50 Kbps < DL TP < 250 Kbps (%)                 754
250 Kbps < DL TP < 1 Mbps (%)                  754
DL TP > 1 Mbps (%)             

In [27]:
# drop columns with more than 30% missing values since filling this will greatly affect the dataset
df_clean = db.drop(['TCP DL Retrans. Vol (Bytes)', 'TCP UL Retrans. Vol (Bytes)', 'HTTP DL (Bytes)', 'HTTP UL (Bytes)', 'Nb of sec with 125000B < Vol DL', 'Nb of sec with 1250B < Vol UL < 6250B', 'Nb of sec with 31250B < Vol DL < 125000B', 'Nb of sec with 37500B < Vol UL', 'Nb of sec with 6250B < Vol DL < 31250B', 'Nb of sec with 6250B < Vol UL < 37500B',], axis=1)
df_clean.shape

(150001, 45)

In [36]:
# dropping rows for columns with Unidue identifiers for customer data since we cannot fill missing values with existing data
df_clean.dropna(subset = ['MSISDN/Number'], inplace=True)

In [38]:
# Checking for missing values and then shape
df_clean.isna().sum()
df_clean.shape

(148935, 45)

In [39]:
# Checking for missing values
df_clean.isna().sum()

Bearer Id                         429
Start                               0
Start ms                            0
End                                 0
End ms                              0
Dur. (ms)                           0
IMSI                                0
MSISDN/Number                       0
IMEI                                0
Last Location Name                589
Avg RTT DL (ms)                     0
Avg RTT UL (ms)                     0
Avg Bearer TP DL (kbps)             0
Avg Bearer TP UL (kbps)             0
DL TP < 50 Kbps (%)               720
50 Kbps < DL TP < 250 Kbps (%)    720
250 Kbps < DL TP < 1 Mbps (%)     720
DL TP > 1 Mbps (%)                720
UL TP < 10 Kbps (%)               767
10 Kbps < UL TP < 50 Kbps (%)     767
50 Kbps < UL TP < 300 Kbps (%)    767
UL TP > 300 Kbps (%)              767
Activity Duration DL (ms)           0
Activity Duration UL (ms)           0
Dur. (ms).1                         0
Handset Manufacturer                0
Handset Type

Filling Missing Values

In [41]:
# fill missing with ffill method for columns ('Nb of sec with Vol DL < 6250B','Nb of sec with Vol UL < 1250B', 'Avg RTT DL (ms)', 'Avg RTT UL (ms)')

def fix_missing_ffill(df, col):
    df[col] = df[col].fillna(method='ffill')
    return df[col]


def fix_missing_bfill(df, col):
    df[col] = df[col].fillna(method='bfill')
    return df[col]

df_clean['Nb of sec with Vol DL < 6250B'] = fix_missing_ffill(df_clean, 'Nb of sec with Vol DL < 6250B')
df_clean['Nb of sec with Vol UL < 1250B'] = fix_missing_ffill(df_clean, 'Nb of sec with Vol UL < 1250B')
df_clean['Avg RTT DL (ms)'] = fix_missing_ffill(df_clean, 'Avg RTT DL (ms)')
df_clean['Avg RTT UL (ms)'] = fix_missing_ffill(df_clean, 'Avg RTT UL (ms)')

# fill 'Handset Manufacturer' column with mode 
df_clean['Handset Manufacturer'] = df_clean['Handset Manufacturer'].fillna(df_clean['Handset Manufacturer'].mode()[0])
# fill 'Handset Type' column with mode 
df_clean['Handset Type'] = df_clean['Handset Type'].fillna(df_clean['Handset Type'].mode()[0])
df_clean.describe()

Bearer Id       Start ms         End ms     Dur. (ms)          IMSI  \
count  1.485060e+05  148935.000000  148935.000000  1.489350e+05  1.489350e+05   
mean   1.014676e+19     499.246732     498.675442  1.048702e+05  2.082016e+14   
std    2.893262e+18     288.555630     288.089136  8.106343e+04  1.522166e+10   
min    6.917538e+18       0.000000       0.000000  7.142000e+03  2.040471e+14   
25%    7.349883e+18     250.000000     251.000000  5.788350e+04  2.082014e+14   
50%    7.349883e+18     499.000000     500.000000  8.639900e+04  2.082015e+14   
75%    1.304243e+19     749.000000     750.000000  1.327000e+05  2.082018e+14   
max    1.318654e+19     999.000000     999.000000  1.859336e+06  2.082522e+14   

       MSISDN/Number          IMEI  Avg RTT DL (ms)  Avg RTT UL (ms)  \
count   1.489350e+05  1.489350e+05    148935.000000    148935.000000   
mean    4.188282e+10  4.849493e+13       102.357089        17.125498   
std     2.447443e+12  2.242829e+13       554.550371        84.596474   
min     3.360100e+10  4.400152e+11         0.000000         0.000000   
25%     3.365130e+10  3.546071e+13        31.000000         2.000000   
50%     3.366371e+10  3.572221e+13        44.000000         5.000000   
75%     3.368349e+10  8.611970e+13        67.000000        13.000000   
max     8.823971e+14  9.900120e+13     96923.000000      7120.000000   

       Avg Bearer TP DL (kbps)  ...  Youtube DL (Bytes)  Youtube UL (Bytes)  \
count            148935.000000  ...        1.489350e+05        1.489350e+05   
mean              13286.171390  ...        1.163345e+07        1.100867e+07   
std               23978.591234  ...        6.711384e+06        6.345428e+06   
min                   0.000000  ...        5.300000e+01        1.050000e+02   
25%                  43.000000  ...        5.831331e+06        5.516710e+06   
50%                  63.000000  ...        1.161638e+07        1.101329e+07   
75%               19681.000000  ...        1.744571e+07        1.651442e+07   
max              378160.000000  ...        2.325910e+07        2.201196e+07   

       Netflix DL (Bytes)  Netflix UL (Bytes)  Gaming DL (Bytes)  \
count        1.489350e+05        1.489350e+05       1.489350e+05   
mean         1.162756e+07        1.100017e+07       4.220256e+08   
std          6.724686e+06        6.358918e+06       2.439763e+08   
min          4.200000e+01        3.500000e+01       2.516000e+03   
25%          5.779947e+06        5.474451e+06       2.104587e+08   
50%          1.164269e+07        1.099563e+07       4.234208e+08   
75%          1.747072e+07        1.650429e+07       6.331620e+08   
max          2.325919e+07        2.201196e+07       8.434419e+08   

       Gaming UL (Bytes)  Other DL (Bytes)  Other UL (Bytes)  \
count       1.489350e+05      1.489350e+05      1.489350e+05   
mean        8.289085e+06      4.211473e+08      8.263189e+06   
std         4.782090e+06      2.431970e+08      4.768708e+06   
min         5.900000e+01      3.290000e+03      1.480000e+02   
25%         4.130162e+06      2.102119e+08      4.145072e+06   
50%         8.291222e+06      4.218716e+08      8.266089e+06   
75%         1.243184e+07      6.316602e+08      1.238204e+07   
max         1.655879e+07      8.434425e+08      1.655882e+07   

       Total UL (Bytes)  Total DL (Bytes)  
count      1.489350e+05      1.489350e+05  
mean       4.111764e+07      4.546251e+08  
std        1.127933e+07      2.441528e+08  
min        2.866892e+06      7.114041e+06  
25%        3.321774e+07      2.430996e+08  
50%        4.114101e+07      4.558708e+08  
75%        4.903248e+07      6.656577e+08  
max        7.833131e+07      9.029696e+08  

[8 rows x 40 columns]

SELECT COLUMNS OF INTEREST AND CALCULATE SUMMARY STATISTICS


In [46]:
#df_clean[['MSISDN/Number', 'Dur. (ms)', 'Social Media DL (Bytes)', 'Social Media UL (Bytes)', 'Google DL (Bytes)', 'Google UL (Bytes)', 'Email DL (Bytes)', 'Email UL (Bytes)', 'Youtube DL (Bytes)', 'Youtube UL (Bytes)', 'Netflix DL (Bytes)', Netflix UL (Bytes)', 'Gaming DL (Bytes)', 'Gaming UL (Bytes)', 'Other DL (Bytes)', 'Other UL (Bytes)', 'Total UL (Bytes)', 'Total DL (Bytes)']]

In [50]:
#aggregate of xDR sessions
xDR_Sessions = df_clean[['Dur. (ms)', 'Dur. (ms).1']]
xDR_Sessions.aggregate([np.sum, 'mean'], axis=0)

Dur. (ms)   Dur. (ms).1
sum   1.561884e+10  1.561892e+13
mean  1.048702e+05  1.048707e+08

In [49]:
# aggregate of the total download and upload data
Total_Data = df_clean[['Total UL (Bytes)', 'Total DL (Bytes)']]
Total_Data.aggregate([np.sum, 'mean'], axis=0)

Total UL (Bytes)  Total DL (Bytes)
sum       6.123856e+12      6.770959e+13
mean      4.111764e+07      4.546251e+08

In [51]:
#aggregate of social media sessions
social_media = df_clean[['Social Media DL (Bytes)', 'Social Media UL (Bytes)']]
social_media.aggregate([np.sum, 'mean'], axis=0)

Social Media DL (Bytes)  Social Media UL (Bytes)
sum              2.673623e+11             4.903196e+09
mean             1.795161e+06             3.292172e+04

In [52]:
#aggregate of google sessions
google = df_clean[['Google DL (Bytes)', 'Google UL (Bytes)']]
google.aggregate([np.sum, 'mean'], axis=0)

Google DL (Bytes)  Google UL (Bytes)
sum        8.566107e+11       3.062427e+11
mean       5.751574e+06       2.056217e+06

In [53]:
#aggregate of email sessions
email = df_clean[['Email DL (Bytes)', 'Email UL (Bytes)']]
email.aggregate([np.sum, 'mean'], axis=0)

Email DL (Bytes)  Email UL (Bytes)
sum       2.668571e+11      6.961065e+10
mean      1.791769e+06      4.673895e+05

In [54]:
#aggregate of youtube sessions
youtube = df_clean[['Youtube DL (Bytes)', 'Youtube UL (Bytes)']]
youtube.aggregate([np.sum, 'mean'], axis=0)

Youtube DL (Bytes)  Youtube UL (Bytes)
sum         1.732628e+12        1.639576e+12
mean        1.163345e+07        1.100867e+07

In [56]:
#aggregate of netflix sessions
netflix = df_clean[['Netflix DL (Bytes)', 'Netflix UL (Bytes)']]
netflix.aggregate([np.sum, 'mean'], axis=0)

Netflix DL (Bytes)  Netflix UL (Bytes)
sum         1.731750e+12        1.638310e+12
mean        1.162756e+07        1.100017e+07

In [57]:
#aggregate of gaming sessions
gaming = df_clean[[ 'Gaming DL (Bytes)', 'Gaming UL (Bytes)']]
gaming.aggregate([np.sum, 'mean'], axis=0)

Gaming DL (Bytes)  Gaming UL (Bytes)
sum        6.285438e+13       1.234535e+12
mean       4.220256e+08       8.289085e+06

In [58]:
#aggregate of other sessions
other = df_clean[['Other DL (Bytes)', 'Other UL (Bytes)']]
other.aggregate([np.sum, 'mean'], axis=0)

Other DL (Bytes)  Other UL (Bytes)
sum       6.272357e+13      1.230678e+12
mean      4.211473e+08      8.263189e+06

ASSIGNMENT 1.2